### PR TITLE
log streamInfo when kinesis video stream is created

### PR DESCRIPTION
**Pull request by chehefen**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/client/src/Stream.c [hunk 1]
```diff
@@ -3245,3 +3246,46 @@ STATUS resetStream(PKinesisVideoStream pKinesisVideoStream)
     return retStatus;
 }
 
+VOID logStreamInfo(PStreamInfo pStreamInfo) {
```
Comment (MushMal) [inline src/client/src/Stream.c]: { -> \n{

File: src/client/src/Stream.c [hunk 2]
```diff
@@ -3245,3 +3246,46 @@ STATUS resetStream(PKinesisVideoStream pKinesisVideoStream)
...
+            streamingType = (PCHAR) "REALTIME";
+            break;
+        case STREAMING_TYPE_OFFLINE:
+            streamingType = (PCHAR) "OFFLINE";
+            break;
+        case STREAMING_TYPE_NEAR_REALTIME:
+            streamingType = (PCHAR) "NEAR_REALTIME";
+            break;
+    }
+
```
Comment (MushMal) [inline src/client/src/Stream.c]: Remove extra \n

File: src/client/src/Stream.c [hunk 3]
```diff
@@ -3245,3 +3246,46 @@ STATUS resetStream(PKinesisVideoStream pKinesisVideoStream)
...
+                  "Max latency (100ns): %" PRIu64 "\n\t"
+                  "Fragment duration (100ns): %" PRIu64 "\n\t"
+                  "Key frame fragmentation: %s\n\t"
+                  "Absolute frame timecode: %s\n\t"
+                  "Buffer duration (100ns): %" PRIu64 "\n\t"
+                  "Replay duration (100ns): %" PRIu64 "\n\t"
+                  "Connection Staleness duration (100ns): %" PRIu64 "\n\t"
+                  "Frame order mode: %u",
+          pStreamInfo->name,
+          streamingType,
```
Comment (MushMal) [inline src/client/src/Stream.c]: Would be nice to stringify this. Have a small macro that returns a pre-canned string for Realtime/Offline

File: src/client/src/Stream.c [hunk 4]
```diff
@@ -3245,3 +3246,46 @@ STATUS resetStream(PKinesisVideoStream pKinesisVideoStream)
...
+          pStreamInfo->streamCaps.maxLatency,
+          pStreamInfo->streamCaps.fragmentDuration ? "Yes" : "No",
+          pStreamInfo->streamCaps.keyFrameFragmentation ? "Yes" : "No",
+          pStreamInfo->streamCaps.absoluteFragmentTimes ? "Yes" : "No",
+          pStreamInfo->streamCaps.bufferDuration,
+          pStreamInfo->streamCaps.replayDuration,
+          pStreamInfo->streamCaps.connectionStalenessDuration,
+          pStreamInfo->streamCaps.frameOrderingMode);
```
Comment (MushMal) [inline src/client/src/Stream.c]: I think you should dump out some of the other fields that might be important
contentType,
frameTimecodes,
nalAdaptationFlags,
avgBandwidthBps, --> this and the next one would give some idea what the size of the view would be but we already print out the size
frameRate,
replayDuration,
connectionStalenessDuration,
segmentUuid

Most importantly, we might want to print out also the track infos

Comment (MushMal) [inline src/client/src/Stream.c]: Danger: frameOrderingMode is a V1 feature. Make sure you have the V1 in the version before dereferencing this

File: src/client/src/Stream.c [hunk 5]
```diff
@@ -3245,3 +3246,46 @@ STATUS resetStream(PKinesisVideoStream pKinesisVideoStream)
...
+        case STREAMING_TYPE_OFFLINE:
+            streamingType = (PCHAR) "OFFLINE";
+            break;
+        case STREAMING_TYPE_NEAR_REALTIME:
+            streamingType = (PCHAR) "NEAR_REALTIME";
+            break;
+    }
+
+
+    DLOGD("Kinesis Video Stream Info:\n\t"
```
Comment (MushMal) [inline src/client/src/Stream.c]: You might want to split this into a couple of DLOGDs as it can overflow the buffer and print out only portion

**Comment by alexw91** [comment]
# [Codecov](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/15?src=pr&el=h1) Report
> Merging [#15](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/15?src=pr&el=desc) into [master](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/commit/ee276ebff6a2e21730b1e9412d20805a1e72eccf&el=desc) will **decrease** coverage by `3.46%`.
> The diff coverage is `95.83%`.

[Impacted file tree graph](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/15?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master      #15      +/-   ##
==========================================
- Coverage   85.63%   82.17%   -3.47%     
==========================================
  Files          45       47       +2     
  Lines        7985     8252     +267     
==========================================
- Hits         6838     6781      -57     
- Misses       1147     1471     +324     
```


| [Impacted Files](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/15?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [src/utils/src/TimerQueue.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/15/diff?src=pr&el=tree#diff-c3JjL3V0aWxzL3NyYy9UaW1lclF1ZXVlLmM=) | `0.00% <0.00%> (-82.81%)` | :arrow_down: |
| [src/utils/src/Thread.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/15/diff?src=pr&el=tree#diff-c3JjL3V0aWxzL3NyYy9UaHJlYWQuYw==) | `52.54% <0.00%> (-10.17%)` | :arrow_down: |
| [src/client/src/InputValidator.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/15/diff?src=pr&el=tree#diff-c3JjL2NsaWVudC9zcmMvSW5wdXRWYWxpZGF0b3IuYw==) | `84.24% <0.00%> (-4.00%)` | :arrow_down: |
| [src/mkvgen/src/MkvGenerator.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/15/diff?src=pr&el=tree#diff-c3JjL21rdmdlbi9zcmMvTWt2R2VuZXJhdG9yLmM=) | `90.85% <0.00%> (-0.04%)` | :arrow_down: |
| [src/utils/src/Tags.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/15/diff?src=pr&el=tree#diff-c3JjL3V0aWxzL3NyYy9UYWdzLmM=) | `100.00% <0.00%> (ø)` | |
| [src/utils/src/Hex.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/15/diff?src=pr&el=tree#diff-c3JjL3V0aWxzL3NyYy9IZXguYw==) | `42.55% <0.00%> (ø)` | |
| [src/utils/src/Base64.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/15/diff?src=pr&el=tree#diff-c3JjL3V0aWxzL3NyYy9CYXNlNjQuYw==) | `0.00% <0.00%> (ø)` | |
| [src/heap/src/Common.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/15/diff?src=pr&el=tree#diff-c3JjL2hlYXAvc3JjL0NvbW1vbi5j) | `99.11% <0.00%> (+0.01%)` | :arrow_up: |
| [src/heap/src/AivHeap.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/15/diff?src=pr&el=tree#diff-c3JjL2hlYXAvc3JjL0FpdkhlYXAuYw==) | `94.08% <0.00%> (+0.11%)` | :arrow_up: |
| [src/heap/src/HybridHeap.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/15/diff?src=pr&el=tree#diff-c3JjL2hlYXAvc3JjL0h5YnJpZEhlYXAuYw==) | `85.45% <0.00%> (+0.20%)` | :arrow_up: |
| ... and [3 more](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/15/diff?src=pr&el=tree-more) | |

------

[Continue to review full report at Codecov](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/15?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/15?src=pr&el=footer). Last update [ee276eb...5e953be](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-pic/pull/15?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Pull request merged**